### ISSUE TITLE
Using createSPList on existing Site Pages list results in newly created list

### ISSUE THREAD

**Issue by Robert1976** [SharePoint/sp-dev-docs#3066]
- [ ] Question
- [ ] Typo
- [x] Bug
- [ ] Additional article idea

I am using the createSPList verb to add site columns to an existing library (in my case the 'Site Pages' list with template 119). As value for the parameter listName I use 'Site Pages'.

My SiteScript works fine as long as I choose English as the site language at site creation time. However, when I choose for example Dutch, a second 'Site Pages' library is created next to the out of the box 'Site Pages' library (I end up with 2 Site Page libraries with template 119). Expected behavior is that my sitescript works with any language an enduser can choose at self service site creation.

Thank you.

**Comment by iamguptarishi** [comment]
This seems to be a bug. I tried with ListName SitePages and subaction as Title = Site Pages. It created 2 different library with same Name and the new library has URL as SitePages1.

**Comment by jstrugnell** [comment]
I just experienced the same. I used the createSPList verb with a listName property of "ClientDetails". I then used the "setTitle" subaction to change the name to "Client Details". When reapplying the associated site design it keeps on creating a new list, with an incrementing number added to the URL. I understand why this is happening. Perhaps we could have Internal/Display Name properties so that we don't need to "rename" the list after creation? Obvious workaround for now is to not use the setTitle subaction within createSPList.

**Comment by sebastienlevert** [comment]
We are experimenting the same. @seansquires is this something you guys are aware of? It looks like me might be blocked by this right now. Thanks!

**Comment by bcameron1231** [comment]
updating - seems like a duplicate of #2993 .

**Comment by pkmelee337** [comment]
@bcameron1231 No, this is not a duplicate. I tried using listName "SitePages" and listName "Site Pages" (and template 119) but still 2 libraries are created. In my opinion the way this should work is checking if a list with the same url as the listName/listUrl parameter from the site script exists. (optionaly with a listTitle parameter) If so we should have a boolean parameter "updateExistingList" or something which, if true off course, updates the list or not. 

Currently this won't work on any list if we have to support list titles in multiple languages and want to use the same url for each language, because I'm not able to change the title. If I do, on each re-apply of the sitedesign a new list is created.

Another small and nice addition would be if we could specify is the list is created under /lists or not. :-)

**Comment by pkmelee337** [comment]
Any updates on this one?

**Comment by Renevdo** [comment]
Any updates on this?

**Comment by seansquires** [comment]
Adding @melitoSP for notice/follow-up
